lib/mixins/template-stamp.html

```diff
@@ -462,6 +127,181 @@
+       *     },
+       *     ...
+       *   ]
+       *
+       * @param {HTMLTemplateElement} template
+       * @param {boolean=} stripWhiteSpace
+       * @param {Document=} ownerDocument
+       * @return {Array<Object>}
+       */
+      static _prepareTemplate(template, stripWhiteSpace, ownerDocument) {
+        // TODO(kschaaf): File issue and/or remove when fixed
+        // hold a reference to content as _content to prevent odd Chrome gc issue
+        // nested templates also may receive their content as _content
+        let content = (template._content = template._content || template.content);
+        // since a template may be re-used, memo-ize notes.
+        if (!content._notes) {
+          let notes = content._notes = [];
+          notes.ownerDocument = ownerDocument || template.ownerDocument;
+          notes.stripWhiteSpace = stripWhiteSpace || template.hasAttribute('strip-whitespace');
+          this._parseTemplateNode(content, {notes});
+        }
+        return content._notes;
+      }
+
+      // add annotations gleaned from subtree at `node` to `notes`
+      static _parseTemplateNode(node, note) {
+        let noted;
+        if (node.localName == 'template' &&
+          !node.hasAttribute('preserve-content')) {
+          noted |= this._parseTemplate(node, note);
+        } else if (node.localName === 'slot') {
+          node._hasInsertionPoint = true;
+        }
+        if (node.firstChild) {
+          noted |= this._parseTemplateChildNodes(node, note);
+        }
+        if (node.attributes) {
+          noted |= this._parseTemplateNodeAttributes(node, note);
+        }
+        return noted;
+      }
+
+      // add annotations gleaned from children of `root` to `list`, `root`'s
+      // `note` is supplied as it is the note.parent of added annotations
+      static _parseTemplateChildNodes(root, note) {
+        let node = root.firstChild;
+        if (node) {
+          let i = 0;
+          while (node) {
+            // Wrap templates
+            if (node.localName == 'template') {
+              node = wrapTemplateExtension(node);
+            }
+            // collapse adjacent textNodes: fixes an IE issue that can cause
+            // text nodes to be inexplicably split =(
+            // note that root.normalize() should work but does not so we do this
+            // manually.
+            let next = node.nextSibling;
+            if (node.nodeType === Node.TEXT_NODE) {
+              let n = next;
+              while (n && (n.nodeType === Node.TEXT_NODE)) {
+                node.textContent += n.textContent;
+                next = n.nextSibling;
+                root.removeChild(n);
+                n = next;
+              }
+              // optionally strip whitespace
+              if (note.notes.stripWhiteSpace && !node.textContent.trim()) {
+                root.removeChild(node);
+                // decrement index since node is removed
+                i--;
```

Comment:
seems like this can just `continue` and remove the `if (node.parentNode)` check below